src/DynamoCore/Engine/MigrateFormulaToDS.cs

```diff
@@ -0,0 +1,101 @@
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Text.RegularExpressions;
+using ProtoCore;
+using ProtoCore.AST.AssociativeAST;
+using ProtoCore.SyntaxAnalysis;
+using ProtoCore.Utils;
+
+namespace Dynamo.Engine
+{
+    internal class MigrateFormulaToDS : AstReplacer
+    {
+        private static readonly TextInfo textInfo = CultureInfo.InvariantCulture.TextInfo;
+
+        /// <summary>
+        /// Regex to match "if(<condition>, <true>, <false>)" pattern in Formula node and extract the parts of the
+        /// "if" statement to construct the corresponding conditional AST in DS. This is because the DS parser does not
+        /// recognize this "if" syntax.
+        /// </summary>
+        private const string ifCond = @"(if\s*)\(\s*(.*?)\s*\)$";
+        private static readonly Regex ifRgx = new Regex(ifCond, RegexOptions.IgnoreCase);
+
+
+        internal string ConvertFormulaToDS(string formula)
+        {
+            CodeBlockNode cbn = null;
+            var match = ifRgx.Match(formula);
```

Comment:
is it simpler to use the DS parser to check for conditional statements - or theres a high chance of failure trying to parse the entire formula string?

Reply:
This syntax for the `if` statement will fail to parse using the DS parser, which is why I'm splitting it up using regex into parts that can be sent individually to the DS parser.

Reply:
When the parser fails, it will throw an exception and the entire deserialization of the formula node will be skipped.